### PR TITLE
Use actions-ecosystem/action-add/remove-labels

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Instead of using a custom implementation of adding/removing labels, should we use the third-party action (https://github.com/actions-ecosystem/action-add-labels, https://github.com/actions-ecosystem/action-remove-labels)?

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [ ] ~Add / update necessary tests?~
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Comment by jezdez** [comment]
That's reasonable, https://github.com/actions-ecosystem is relatively well-known and trust-worthy. But as the GitHub docs mention, we should "harden" our use of GHA by referring to commit, not tags.

File: check-cla/action.yml [hunk 1]
```diff
@@ -76,31 +76,21 @@ runs:
           core.setOutput('contributors', contributors);
           core.setOutput('hasSigned', hasSigned);
 
-    # add cla-signed label if actor has already signed
+    # add [cla-signed] label if actor has already signed
     - name: Add label
-      uses: actions/github-script@v6
```
Comment (jezdez) [inline check-cla/action.yml:81]: This would need to be updated to a commit

File: check-cla/action.yml [hunk 2]
```diff
@@ -76,31 +76,21 @@ runs:
...
-              issue_number: context.issue.number,
-              labels: ['${{ inputs.label }}']
-            })
+        github_token: ${{ inputs.token }}
+        labels: ${{ inputs.label }}
 
-    # remove cla-signed label if actor has not signed yet
+    # remove [cla-signed] label if actor has not signed yet
     - name: Remove label
-      uses: actions/github-script@v6
```
Comment (jezdez) [inline check-cla/action.yml:94]: Same, commit needed here.

**Review by jezdez — CHANGES_REQUESTED** [review]
I wonder if we should only allow actions from certain GitHub orgs in the conda organization settings FWIW. That may put a scope to the risk maybe?

**Comment by dbast** [comment]
related to that: Would be interesting to figure out how to update the pinned sha1s to newer tagged versions via dependabot/renovate ... dependabot seems to only supports .github/workflow/*.yml files... no composite action.yml support even though the syntax and versioning is the same.. not sure about renovate

**Review by dbast — APPROVED** [review]

**Review by dbast — APPROVED** [review]

**Pull request merged**